algorithm.py

```diff
@@ -39,9 +41,12 @@
     ...
 
 
-Each function returns a :py:class:`dict`, which is used to update the state of
-the current run of the algorithm. Names from the state dictionary are made
-available to downstream functions via :py:mod:`dependency_injection`. Now
+Each function returns a :py:class:`dict`, which updates the state of
```

Comment:
For some reason it sounded too low level without real explanation. It is kind of obvious that returned dict can not update the state itself, so I decided to leave it out. But we can mention that it is passed to `state.update` automatically.